File: Level_4(C++,Java)/java/Static/Countable/Countable.java
public class Countable implements Cloneable {
    public static int total = 0;
    
    public Countable() {
        total += 1;
    }
    
    public Countable(Countable copy) {
        total += 1;
    }
    
    public Countable clone() throws CloneNotSupportedException {
        return (Countable) super.clone();
    }
}

File: Level_4(C++,Java)/C++/static/Countable/Countable.cpp
#include "Countable.h"

int Countable::total = 0;

Countable::Countable() {
    total += 1;
}

Countable::Countable(const Countable& copy) {
    total += 1;
}

Countable::~Countable() {
    total -= 1;
}

Countable& Countable::operator=(const Countable& copy) {    
    return *this;
}
